### PR TITLE
feat: Rendering the generated code before downloading the file

### LINKED ISSUE #324: Window or tab to view the code before downloading the file
## Feature Request

Is your feature request related to an existing [Issue](https://www.github.com/Auto-DL/Generator/issues) or a new implementation?

No.

### Feature Description

Add a window or tab to view the code before downloading to check if everything is alright. The user can also opt to copy the code instead of downloading the file. Since the user may be skeptical of downloading files as google chrome categorizes the file as malicious.

### Possible Solution

Generate the code that is in the file and show the user.  
Better solution is keep a option to view the code at any time during the ML pipeline by dynamically generating it. ( Could be an issue in itself )

### PR DESCRIPTION
**Pull request by AmitGupta7580**
# Pull Request

## What does this PR do?
Fixes #324

image

## What part does this affect?
- [x] FrontEnd.
- [ ] BackEnd.
- [ ] Documentation.
- [ ] Other. (Please specify below)

## Before submitting
- [x] Was this discussed/approved via a GitHub issue or slack?
- [x] Did you read the [contributor guideline](https://github.com/Auto-DL/Auto-DL/blob/v1-beta/CONTRIBUTING.md)?
- [x] Did you ensure that there aren't any other open [Pull Requests](https://www.github.com/Auto-DL/Generator/pulls) for the same update/change?
- [x] Did you make sure the title is self-explanatory and the description concisely explains the PR?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure the code is clean and docstrings have been added or updated as required?
- [x] Did you make sure the code is linted/formatted locally prior to submission? (using `black` and/or `prettier`)
- [x] Did you make sure to update the documentation with your changes? (if necessary)

## PR review
Anyone in the community is free to review the PR once the tests have passed.

Thank you for contributing to [AutoDL](https://github.com/Auto-DL/Auto-DL). We look forward to your continued support.

### CODE REVIEW

**Review by ADI10HERO — COMMENTED** [review]
LGTM

**Comment by ADI10HERO** [comment]
@McTechie  could you please review as well? :)

File: FrontEndApp/v1-react/src/Components/MainApp/Step2.js [hunk 1]
```diff
@@ -2801,31 +2809,7 @@ function Step2() {
       <Dialog onClose={handleCloseModal} open={openModal}>
         <DialogTitle onClose={handleCloseModal}>Code Generated!</DialogTitle>
```
Comment (McTechie) [inline FrontEndApp/v1-react/src/Components/MainApp/Step2.js:2810]:
```suggestion
        <DialogTitle onClose={handleCloseModal}>Great Work!</DialogTitle>
```

File: FrontEndApp/v1-react/src/Components/MainApp/Step2.js [hunk 2]
```diff
@@ -2801,31 +2809,7 @@ function Step2() {
...
-                <Tooltip
-                  title="Exmaple: python3 test.py"
-                  placement="bottom-start"
-                >
-                  <div> Run the code.</div>
-                </Tooltip>
-              </li>
-            </ul>
-          </div>
+          <p style={{whiteSpace: 'pre'}}> { code } </p>
```
Comment (McTechie) [inline FrontEndApp/v1-react/src/Components/MainApp/Step2.js:2812]:
```suggestion
          <div>
            <h3>Generated Code:</h3>
            <p style={{ whiteSpace: 'pre' }}>
              { code }
            </p>
            
            <hr />

            <h3>Instructions:</h3>
            <ul>
              <li>
                Click the "Download Code" button to download the generated code
                to any directory of your choice.
              </li>
              <br></br>
              <Tooltip title="See value of 'base' variable in the python file">
                <li>
                  Make sure you place the data files relative to the downloaded
                  script
                </li>
              </Tooltip>
              <br></br>
              <li>
                <Tooltip
                  title="Exmaple: python3 test.py"
                  placement="bottom-start"
                >
                  <div> Run the code.</div>
                </Tooltip>
              </li>
            </ul>
          </div>
```

**Review by McTechie — CHANGES_REQUESTED** [review]
The functionality looks good to me, just a few UI changes xD

**Comment by AmitGupta7580** [comment]
@ADI10HERO @McTechie I have updated the UI as requested have a look on the changes.

**Review by McTechie — APPROVED** [review]
Looks good to me xD

**Pull request merged**